api/v1alpha1/stresschaos_webhook.go

```diff
@@ -0,0 +1,188 @@
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// +kubebuilder:object:generate=false
+
+// Validateable defines how a resource is validated
+type Validateable interface {
+	Validate(parent *field.Path, errs field.ErrorList) field.ErrorList
+}
+
+// log is for logging in this package.
+var (
+	stressChaosLog = ctrl.Log.WithName("stresschaos-resource")
+)
+
+// SetupWebhookWithManager setup StressChaos's webhook with manager
+func (in *StressChaos) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(in).
+		Complete()
+}
+
+// +kubebuilder:webhook:path=/mutate-pingcap-com-v1alpha1-stresschaos,mutating=true,failurePolicy=fail,groups=pingcap.com,resources=stresschaos,verbs=create;update,versions=v1alpha1,name=mstresschaos.kb.io
+
+var _ webhook.Defaulter = &StressChaos{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (in *StressChaos) Default() {
+	stressChaosLog.Info("default", "name", in.Name)
+	in.Spec.Selector.DefaultNamespace(in.GetNamespace())
+}
+
+// +kubebuilder:webhook:verbs=create;update,path=/validate-pingcap-com-v1alpha1-stresschaos,mutating=false,failurePolicy=fail,groups=pingcap.com,resources=stresschaos,versions=v1alpha1,name=vstresschaos.kb.io
+
+var _ ChaosValidator = &StressChaos{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (in *StressChaos) ValidateCreate() error {
+	stressChaosLog.Info("validate create", "name", in.Name)
+	return in.Validate()
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (in *StressChaos) ValidateUpdate(old runtime.Object) error {
+	stressChaosLog.Info("validate update", "name", in.Name)
+	return in.Validate()
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (in *StressChaos) ValidateDelete() error {
+	stressChaosLog.Info("validate delete", "name", in.Name)
+
+	// Nothing to do?
+	return nil
+}
+
+// Validate validates chaos object
+func (in *StressChaos) Validate() error {
+	root := field.NewPath("stresschaos")
+	errs := field.ErrorList{}
+	errs = in.Spec.Validate(root, errs)
+	errs = append(errs, in.ValidatePodMode(root)...)
+	if len(errs) > 0 {
+		return fmt.Errorf(errs.ToAggregate().Error())
+	}
+	return nil
+}
+
+// ValidatePodMode validates the value with podmode
+func (in *StressChaos) ValidatePodMode(spec *field.Path) field.ErrorList {
+	return ValidatePodMode(in.Spec.Value, in.Spec.Mode, spec.Child("value"))
+}
+
+// ValidateScheduler validates whether scheduler is well defined
+func (in *StressChaos) ValidateScheduler(root *field.Path) field.ErrorList {
+	panic("implement me")
+}
+
+// Validate validates the scheduler and duration
+func (in *StressChaosSpec) Validate(parent *field.Path, errs field.ErrorList) field.ErrorList {
```

Comment:
```suggestion
func (in *StressChaosSpec) Validate(parent *field.Path) field.ErrorList {
```

Why not keep the same style?

Reply:
Fixed